corehq/apps/app_manager/views/modules.py

```diff
@@ -186,11 +186,13 @@ def _report_to_config(report):
     warnings = []
     validity = module.check_report_validity()
 
+    # We're now proactively deleting these references, so after that's been
+    # out for a while, this can be removed (say June 2016 or later)
     if not validity.is_valid:
         module.report_configs = validity.valid_report_configs
```

Comment:
This line here mutates the app to remove the missing report configs, but it's only run on report module pages.  I'm leaving it in now in case there are apps in the wild with this bad state, as this is the easiest way to recover.  This issue fails loudly enough that I'm not concerned about removing this after giving any latent issues a chance to pop up.